kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/core/BroadcastDoubleTensorAlgebra.kt

```diff
@@ -27,16 +27,16 @@ public object BroadcastDoubleTensorAlgebra : DoubleTensorAlgebra() {
         val newThis = broadcast[0]
         val newOther = broadcast[1]
         val resBuffer = DoubleArray(newThis.indices.linearSize) { i ->
-            newThis.mutableBuffer.array()[i] + newOther.mutableBuffer.array()[i]
+            newThis.mutableBuffer[i] + newOther.mutableBuffer[i]
```

Comment:
This could cause performance regression. calling `array()` ensures that we use unboxed access to the array. I recommend to revise DoubleTensor, inherit mutableBuffer and ensure that it is always a DoubleBuffer.